### ISSUE TITLE
rstudio_command_palette_search is slow

### ISSUE THREAD

**Issue by jooyoungseo** [rstudio/rstudio#6961]
### System details

    RStudio Edition : Desktop or Server
    RStudio Version : 1.4.390
    OS Version      : Windows 10 Pro
    R Version       : any

### Steps to reproduce the problem

1. With JAWS screen reader loaded in Chrome, press `Ctrl + Shift + P` to open "Command Palette."
2. Count down how long it takes to bring up the search menu (probably, between 5 and 10 seconds).

### Describe the problem in detail

Compared to VSCode and Google Docs, the search menu is opened very slowly.

Another screen-reader test (e.g., NVDA) gives us a little bit faster performance; however, it is slow basically.

### Describe the behavior you expected

The search menu should be as responsive as possible.

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
